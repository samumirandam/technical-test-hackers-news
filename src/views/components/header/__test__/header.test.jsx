import React from "react";
import { render, screen, fireEvent } from "@utils/test-utils";

import Header from "../index";

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Header {...setupStore} />, {});
};

describe("Test for Header component", () => {
  test("Should render without errors", () => {
    setup();
    expect(screen.getByText("HACKER NEWS")).toBeTruthy();
  });

  test("Should renders the same component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test("Should render click in home icon", () => {
    setup();
    fireEvent.click(screen.getByText("HACKER NEWS"));
    expect(global.window.location.pathname).toEqual("/");
  });
});
