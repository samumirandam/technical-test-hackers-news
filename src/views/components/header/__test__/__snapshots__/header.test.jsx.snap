// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header component Should renders the same component 1`] = `
<div>
  <header
    class="Header"
  >
    <button
      class="Button noStyles"
      type="button"
    >
      <p
        class="Header__logo"
      >
        HACKER NEWS
      </p>
    </button>
  </header>
</div>
`;
